### ISSUE TITLE
Unable to get FLAG column in gain tables

### ISSUE THREAD

**Issue by Mulan-94** [ska-sa/dask-ms#49]
* dask-ms[xarray] '0.2.0-alpha2'
* Python 3.5.2
* Ubuntu 16.04.6 LTS

### Description
Cannot get FLAG column from gain tables using `xds_from_table`
```
tab = xm.xds_from_table(tab_name)[0]
tab.FLAG
*** AttributeError: 'Dataset' object has no attribute 'FLAG'
```
Only tables available:
![image](https://user-images.githubusercontent.com/12611059/63505202-67323d00-c4d3-11e9-85ff-717e62a7048f.png)

**Comment by sjperkins** [comment]
Thanks for the report Lexi!

Could you set the logging level to TRACE or DEBUG (something below) and see if it complains about ignoring a column:

```python
import logging
logging.basicConfig(level=logging.TRACE)
tab = xm.xds_fom_table(tab_name)[0]
```

**Comment by Mulan-94** [comment]
Yes it does. And a couple of other columns:
```
WARNING:daskms.reads:Ignoring column: 'PARAMERR'
Traceback (most recent call last):
  File "/usr/lib/python3.5/cmd.py", line 214, in onecmd
    func = getattr(self, 'do_' + cmd)
AttributeError: 'TerminalPdb' object has no attribute 'do_tab'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/reads.py", line 200, in _dataset_variable_factory
    chunks, exemplar_row)
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/columns.py", line 151, in column_metadata
    % (column, ndim))
daskms.columns.ColumnMetadataError: Unconstrained shapes in column 'PARAMERR' (ndim == -1) are not currently handled
WARNING:daskms.reads:Ignoring column: 'WEIGHT'
Traceback (most recent call last):
  File "/usr/lib/python3.5/cmd.py", line 214, in onecmd
    func = getattr(self, 'do_' + cmd)
AttributeError: 'TerminalPdb' object has no attribute 'do_tab'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/reads.py", line 200, in _dataset_variable_factory
    chunks, exemplar_row)
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/columns.py", line 151, in column_metadata
    % (column, ndim))
daskms.columns.ColumnMetadataError: Unconstrained shapes in column 'WEIGHT' (ndim == -1) are not currently handled
WARNING:daskms.reads:Ignoring column: 'CPARAM'
Traceback (most recent call last):
  File "/usr/lib/python3.5/cmd.py", line 214, in onecmd
    func = getattr(self, 'do_' + cmd)
AttributeError: 'TerminalPdb' object has no attribute 'do_tab'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/reads.py", line 200, in _dataset_variable_factory
    chunks, exemplar_row)
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/columns.py", line 151, in column_metadata
    % (column, ndim))
daskms.columns.ColumnMetadataError: Unconstrained shapes in column 'CPARAM' (ndim == -1) are not currently handled
WARNING:daskms.reads:Ignoring column: 'SNR'
Traceback (most recent call last):
  File "/usr/lib/python3.5/cmd.py", line 214, in onecmd
    func = getattr(self, 'do_' + cmd)
AttributeError: 'TerminalPdb' object has no attribute 'do_tab'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/reads.py", line 200, in _dataset_variable_factory
    chunks, exemplar_row)
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/columns.py", line 151, in column_metadata
    % (column, ndim))
daskms.columns.ColumnMetadataError: Unconstrained shapes in column 'SNR' (ndim == -1) are not currently handled
WARNING:daskms.reads:Ignoring column: 'FLAG'
Traceback (most recent call last):
  File "/usr/lib/python3.5/cmd.py", line 214, in onecmd
    func = getattr(self, 'do_' + cmd)
AttributeError: 'TerminalPdb' object has no attribute 'do_tab'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/reads.py", line 200, in _dataset_variable_factory
    chunks, exemplar_row)
  File "/home/lexya/Documents/HAHA/testenv/lib/python3.5/site-packages/daskms/columns.py", line 151, in column_metadata
    % (column, ndim))
daskms.columns.ColumnMetadataError: Unconstrained shapes in column 'FLAG' (ndim == -1) are not currently handled

```

**Comment by sjperkins** [comment]
Ah OK, thanks very much for reporting this.

I've relaxed the check in this https://github.com/ska-sa/dask-ms/pull/48/commits/601f581412482c93ac2c08b28cacd8aefefef57d.

Could you test the fix out on the branch?

```bash
$ pip install "dask-ms[xarray] @ git+https://github.com/ska-sa/dask-ms.git@fixes-for-0.2.0-alpha2"
```

Could you also mail the table to my SKA account, if it's not to large (a couple of 10's of MB).

**Comment by sjperkins** [comment]
Fixed in #48
